app/upgrades/v16/upgrades.go

```diff
@@ -43,6 +46,22 @@ func CreateUpgradeHandler(
 			logger.Error("failed to enable outposts", "error", err.Error())
 		}
 
+		feeCollectorModuleAccount := ak.GetModuleAccount(ctx, types.FeeCollectorName)
+		if feeCollectorModuleAccount == nil {
+			logger.Error("fee collector module account not found")
+		}
+
+		modAcc, ok := feeCollectorModuleAccount.(*types.ModuleAccount)
+		if !ok {
+			logger.Error("fee collector module account is not a module account")
+		}
+
+		// Create a new FeeCollector module account with the same address and balance as the old one.
+		newFeeCollectorModuleAccount := types.NewModuleAccount(modAcc.BaseAccount, types.FeeCollectorName, types.Burner)
+
+		// Override the FeeCollector module account in the auth keeper.
+		ak.SetModuleAccount(ctx, newFeeCollectorModuleAccount)
+
```

Comment:
nit: I would prefer refactoring this into a function `updateBurnerRole(ctx, ak, types.FeeCollectorName)`